src/olympia/addons/models.py

```diff
@@ -1082,6 +1084,8 @@ def show_adu(self):
     @cached_property
     def current_beta_version(self):
         """Retrieves the latest version of an addon, in the beta channel."""
+        if not waffle.switch_is_active('beta-versions'):
+            return
```

Comment:
Note: since this property is stored in ES and then overwritten in the ES serializer by whatever data was in ES, if we care about its value in the search API (not detail) then we need to reindex every time the switch is toggled. Which is not necessarily a bad thing, as it avoids the cost of checking the waffle for every add-on returned.

Reply:
I believe that this should be ok. :) I don't expect us to switch it off and on too often.